### PR TITLE
Added pull request and issue templates for GitHub.

### PR DESCRIPTION
**Pull request by Clintonio**
This is a first version of our issue and PR templates to help and thank contributors for their work. This will appear in the PR box (like the one I am writing this message into) before the user has selected it for the first time.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Thank you for choosing to reporting an issue with Corda.
```
Comment (joeldudleyr3) [inline .github/ISSUE_TEMPLATE.md]: "to reporting" -> "to report"

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Thank you for choosing to reporting an issue with Corda. 
+
+When reporting an issue please make sure it contains;
+
+* A clear description of the issue
+* Any logs or stack traces that you can provide
```
Comment (joeldudleyr3) [inline .github/ISSUE_TEMPLATE.md]: What is the log is too large?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Thank you for choosing to contribute to Corda. As a contributor please read [CONTRIBUTING](../contributing.md) before
+contributing.
+
+## PR Process
+
+Your PR must be approved by one or more reviewers and all tests must be passed on [TeamCity CI](https://ci.corda.r3cev.com) 
+in order to be merged. 
+
```
Comment (joeldudleyr3) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add information about posting a comment stating that you agree with the DCO/contribution policy.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Once you have submitted a PR you are responsible for keeping it up to date until the time it is merged.
+
+## PR Checklist
+
+1. Ensure you have done any relevant automated and manual testing.
```
Comment (joeldudleyr3) [inline .github/PULL_REQUEST_TEMPLATE.md]: How?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+2. Add your changes to the changelog.
```
Comment (joeldudleyr3) [inline .github/PULL_REQUEST_TEMPLATE.md]: How?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+3. Update any documentation relating to your changes.
```
Comment (joeldudleyr3) [inline .github/PULL_REQUEST_TEMPLATE.md]: How?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
```
Comment (joeldudleyr3) [inline .github/PULL_REQUEST_TEMPLATE.md]: Remind them to post the comment if they're aren't an R3 employee.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (Clintonio) [reply .github/ISSUE_TEMPLATE.md]: 👍

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (Clintonio) [reply .github/ISSUE_TEMPLATE.md]: I'll mention pastebin.com

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (Clintonio) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (Clintonio) [reply .github/PULL_REQUEST_TEMPLATE.md]: This is a checklist, if contributing they'll need to do some prior research. They can ask on Slack, read the docs, etc, but this isn't the place to train people.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (Clintonio) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added the actual file name.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (Clintonio) [reply .github/PULL_REQUEST_TEMPLATE.md]: Mentioned the actual location of the docs. Again though; people need to figure out how to change the docs themselves, this is just a checklist.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (joeldudleyr3) [reply .github/PULL_REQUEST_TEMPLATE.md]: Then how about telling them which docs page to read? Maybe we need a "PRing against Corda" page? And we can link that here?

Comment (Clintonio) [reply .github/PULL_REQUEST_TEMPLATE.md]: How about a testing doc? I can throw a first version in here. I'm a bit blocked at the moment anyway.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+Your PR must be approved by one or more reviewers and all tests must be passed on [TeamCity CI](https://ci.corda.r3cev.com) 
+in order to be merged. 
+
+Once you have submitted a PR you are responsible for keeping it up to date until the time it is merged.
+
+## PR Checklist
+
+1. Ensure any new code is tested as described [here](https://docs.corda.net/testing.html)
+1. Ensure you have done any relevant automated testing and manual testing
```
Comment (joeldudleyr3) [inline .github/PULL_REQUEST_TEMPLATE.md]: Duplicate bullet 1?

File: docs/source/testing.rst [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+**Unit tests**: These are traditional unit tests that should only test a single code unit, typically a method or class.
+
+**Integration tests**: These tests should test the integration of small numbers of units, preferably with mocked out services.
+
+**Smoke tests**: These are full end to end tests which start a full set of Corda nodes and verify broader behaviour.
+
+**Other**: These include tests such as performance tests, stress tests, etc, and may be in an external repo.
+
+These tests are mostly written with JUnit and can be run via ``gradle`` with ``gradlew test integrationTest smokeTest``
+or any combination of these three.
```
Comment (joeldudleyr3) [inline docs/source/testing.rst]: Can you clarify to "three arguments" or something? It confused me for a bit.

File: docs/source/testing.rst [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+**Unit tests**: These are traditional unit tests that should only test a single code unit, typically a method or class.
+
+**Integration tests**: These tests should test the integration of small numbers of units, preferably with mocked out services.
+
+**Smoke tests**: These are full end to end tests which start a full set of Corda nodes and verify broader behaviour.
+
+**Other**: These include tests such as performance tests, stress tests, etc, and may be in an external repo.
+
+These tests are mostly written with JUnit and can be run via ``gradle`` with ``gradlew test integrationTest smokeTest``
```
Comment (joeldudleyr3) [inline docs/source/testing.rst]: Add the mac way of running? "./gradlew ..."

File: docs/source/testing.rst [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+**Sample project** - changes in the ``samples`` project. eg; changing the IRS demo means you should manually test the IRS demo.
+
+**Explorer** - changes to the ``tools/explorer`` project.
+
+**Demobench** - changes to the ``tools/demobench`` project.
+
+How to manually test each of these areas differs and is currently not fully specified. For now the best thing to do is
+ensure the program starts, that you can interact with it, and that no exceptions are generated in normal operation.
+
+TODO: Add instructions on manual testing
```
Comment (joeldudleyr3) [inline docs/source/testing.rst]: Anything about re-running for flaky tests?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 8 above)
Comment (Clintonio) [reply .github/PULL_REQUEST_TEMPLATE.md]: Hah, fixed.

File: docs/source/testing.rst (hunk 10 above)
Comment (Clintonio) [reply docs/source/testing.rst]: Fixed.

File: docs/source/testing.rst (hunk 9 above)
Comment (Clintonio) [reply docs/source/testing.rst]: :+1:

File: docs/source/testing.rst (hunk 11 above)
Comment (Clintonio) [reply docs/source/testing.rst]: For now no; we have no record of which are flaky. This could lead to people assuming _new_ flakiness is fine.

**Pull request merged**